File: helloworld-server/userserver/src/main/java/com/example/helloworld/userserver/member/dto/request/CoupleCreateRequest.java
package com.example.helloworld.userserver.member.dto.request;


import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;

import java.time.LocalDate;

@JsonInclude(JsonInclude.Include.NON_NULL)
@Schema(name="CoupleCreateRequest", description="여성(userA) 본인이 커플 생성")
public record CoupleCreateRequest(
        @Schema(description = "임신 주차", example = "12")
        Integer pregnancyWeek,

        @JsonProperty("due_date")
        @Schema(description = "출산 예정일(yyyy-MM-dd)", example = "2026-02-01")
        LocalDate dueDate,

        @JsonProperty("menstrual_date")
        @Schema(description = "최근 생리 시작일(yyyy-MM-dd)", example = "2025-09-01")
        LocalDate menstrualDate,

        @JsonProperty("is_childbirth")
        @Schema(description = "출산 완료 여부", example = "false")
        Boolean isChildbirth
) {}

File: helloworld-android/app/src/main/java/com/ms/helloworld/dto/request/CoupleCreateRequest.kt
package com.ms.helloworld.dto.request

import com.google.gson.annotations.SerializedName

data class CoupleCreateRequest(
    val pregnancyWeek: Int? = null,
    @SerializedName("due_date") val due_date: String? = null, // "yyyy-MM-dd" format
    @SerializedName("menstrual_date") val menstrual_date: String? = null, // "yyyy-MM-dd" format
    @SerializedName("menstrual_cycle") val menstrual_cycle: Int? = null, // 생리 주기 (일수)
    @SerializedName("is_childbirth") val is_childbirth: Boolean? = null
)